File: examples/02-reliability/python/workflow_time.py
from datetime import timedelta

from temporalio import workflow


@workflow.defn
class TimeWorkflow:
    @workflow.run
    async def run(self) -> None:
        # Replay-safe timer. Records a TimerStarted event and frees the worker
        # thread instead of really sleeping. Never use asyncio.sleep / time.sleep.
        await workflow.sleep(timedelta(hours=6))

        # Replay-safe time source. Never call datetime.now() in a Workflow.
        deadline = workflow.now() + timedelta(days=1)

File: examples/02-reliability/go/workflow_time.go
package reliability

import (
	"time"

	"go.temporal.io/sdk/workflow"
)

func TimeWorkflow(ctx workflow.Context) error {
	// Replay-safe timer. Records a TimerStarted event and frees the worker
	// goroutine instead of really sleeping. Never use time.Sleep in a Workflow.
	_ = workflow.Sleep(ctx, 6*time.Hour)

	// Replay-safe time source. Never call time.Now() in a Workflow.
	deadline := workflow.Now(ctx).Add(24 * time.Hour)
	_ = deadline
	return nil
}
